### PR TITLE
[SearchSnippet][Part-2] Create SearchSnippet data handler

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- #3235 

## Dependency
- https://github.com/linagora/jmap-dart-client/pull/99

### CODE REVIEW

File: lib/features/thread/data/network/thread_api.dart [hunk 1]
```diff
@@ -83,6 +87,86 @@ class ThreadAPI {
...
+    }
+
+    SearchSnippetGetResponse? searchSnippetGetResponse;
+    try {
+      searchSnippetGetResponse = result.parse<SearchSnippetGetResponse>(
+          getSearchSnippetInvocation.methodCallId,
+          SearchSnippetGetResponse.fromJson);
+    } catch (e) {
+      logError('ThreadAPI::searchEmails:searchSnippetGetResponse: Exception = $e');
+    }
```
Comment (dab246) [inline lib/features/thread/data/network/thread_api.dart:163]: Should separate function to handle it. also write  write unit for it

File: test/features/thread/domain/usecases/search_more_email_interactor_test.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            [searchEmail.toPresentationEmail(
+              searchSnippetSubject: searchEmail.searchSnippetSubject,
+              searchSnippetPreview: searchEmail.searchSnippetPreview,
+            )]
+          ),
+        ]),
+      );
+    });
+
+    test(
```
Comment (dab246) [inline test/features/thread/domain/usecases/search_more_email_interactor_test.dart:62]: check again test failed

File: test/features/thread/domain/usecases/search_email_interactor_test.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import '../../../../fixtures/account_fixtures.dart';
+import '../../../../fixtures/session_fixtures.dart';
+import 'search_email_interactor_test.mocks.dart';
+
+@GenerateNiceMocks([MockSpec<ThreadRepository>()])
+void main() {
+  final threadRepository = MockThreadRepository();
+  final searchEmailInteractor = SearchEmailInteractor(threadRepository);
+
+  group('search email interactor test:', () {
```
Comment (dab246) [inline test/features/thread/domain/usecases/search_email_interactor_test.dart:21]: also write unit test for case: SearchSnippet failed , list email search are list email of `Email/get`

File: lib/features/thread/data/network/thread_api.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/features/thread/data/network/thread_api.dart]: Done

File: test/features/thread/domain/usecases/search_more_email_interactor_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/features/thread/domain/usecases/search_more_email_interactor_test.dart:66]: Done

File: test/features/thread/domain/usecases/search_email_interactor_test.dart (hunk 3 above)
Comment (tddang-linagora) [reply test/features/thread/domain/usecases/search_email_interactor_test.dart:21]: Done

File: test/features/mailbox_dashboard/domain/usecases/quick_search_email_interactor_test.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  group('quick search email interactor test:', () {
+    test(
+      'should return list of presentation emails '
+      'when threadRepository.searchEmails returns list of emails',
+    () async {
+      // arrange
+      final searchEmail = SearchEmail(
+        searchSnippetSubject: 'searchSnippetSubject',
+        searchSnippetPreview: 'searchSnippetPreview',
+      );
```
Comment (dab246) [inline test/features/mailbox_dashboard/domain/usecases/quick_search_email_interactor_test.dart:30]: IMO, id is required.

```suggestion
      final searchEmail = SearchEmail(
        id: EmailId(Id('abc')),
        searchSnippetSubject: 'searchSnippetSubject',
        searchSnippetPreview: 'searchSnippetPreview',
      );
```

Comment (tddang-linagora) [reply test/features/mailbox_dashboard/domain/usecases/quick_search_email_interactor_test.dart]: Done

File: lib/features/thread/domain/model/search_emails_response.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        searchSnippet)));
+
+    return emailList?.map((email) {
+      final searchSnippet = mapSearchSnippet[email.id];
+      return SearchEmail.fromEmail(
+        email,
+        searchSnippetSubject: searchSnippet?.subject,
+        searchSnippetPreview: searchSnippet?.preview);
+    }).toList();
+  }
```
Comment (hoangdat) [inline lib/features/thread/domain/model/search_emails_response.dart:34]: should we test this function to make sure: email list map with correct search snippet.

**Review by hoangdat — DISMISSED** [review]
please resolve conflict

**Review by hoangdat — APPROVED** [review]

**Comment by dab246** [comment]
- [ ] Upgrade new version for `jmap_dart_client`

**Review by dab246 — APPROVED** [review]
LGTM

**Pull request merged**